File: matlab/samples/multiple_sensors.m
%%
% Variables
capture_path = common.get_sample_capture_path();

%%
% Initialize
if cepton_sdk.is_initialized()
    cepton_sdk.deinitialize();
end
options = struct();
options.capture_path = capture_path;
cepton_sdk.initialize(options);

% Get sensors
sensors_dict = cepton_sdk.get_sensors();

%% Get points
listener = cepton_sdk.FramesListener();
points_dict = listener.get_points();
delete(listener);
points_list = values(points_dict);
points_list = points_list{1};
points = points_list{1};

% Plot
cepton_sdk.plot_points(points);

File: python/samples/multiple_sensors.py
#!/usr/bin/env python3
"""
Sample script for getting points from multiple sensors simultaneously.
"""

import pprint

import cepton_sdk
import cepton_sdk.plot
from common import *

if __name__ == "__main__":
    # Variables
    capture_path = get_sample_capture_path()

    # Initialize
    cepton_sdk.initialize(capture_path=capture_path, enable_wait=True)

    # Get sensors
    sensors_dict = cepton_sdk.get_sensors()

    # Get points
    listener = cepton_sdk.FramesListener()
    points_dict = listener.get_points()
    del listener
    points_list = next(iter(points_dict.values()))
    points = points_list[0]

    # Plot
    cepton_sdk.plot.plot_points(points)
